### PR TITLE
feat: Add MongoDb datastore

### PR DESCRIPTION
**Pull request by Elyniss**
~~The code is missing Terraform setup. But it will be handled in another PR.~~

Tests are missing because they're difficult to set up. Will be another PR for that. Tested manually tho.

**Do not deploy as it will break current integrations because of API changes**

Should be only deployed to staging env

### CODE REVIEW

File: terraform/docdb/main.tf [hunk 1]
```diff
@@ -0,0 +1,97 @@
+terraform {
+  required_version = "~> 1.0"
+
+  required_providers {
+    aws = {
+      source  = "hashicorp/aws"
+      version = "~> 3.27"
```
Comment (xav) [inline terraform/docdb/main.tf:7]: Latest is `4.48`

File: terraform/docdb/main.tf [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+    }
+  }
+}
+
+locals {
+  name_prefix     = replace("${var.environment}-${var.app_name}-${var.mongo_name}", "_", "-")
+  master_password = data.aws_secretsmanager_secret_version.master_password.secret_string
+}
+
+data "aws_secretsmanager_secret" "master_password" {
```
Comment (xav) [inline terraform/docdb/main.tf:21]: Do we need the `data` when we already have the `resource`?

File: terraform/docdb/main.tf [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+  arn = aws_secretsmanager_secret.master_password.arn
+}
+
+data "aws_secretsmanager_secret_version" "master_password" {
```
Comment (xav) [inline terraform/docdb/main.tf:25]: Do we need the data when we already have the resource?

File: src/auth/cacao/payload.rs [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+    /// TODO: write valdation
+    pub fn is_valid(&self) -> Result<bool, CacaoError> {
+        Ok(true)
+    }
+
+    pub fn address(&self) -> String {
+        self.iss
+            .split(Self::ISS_DELIMITER)
+            .nth(Self::ISS_POSITION_OF_ADDRESS)
+            .unwrap()
```
[the remark this answers is not in the document]
Comment (Elyniss) [reply src/auth/cacao/payload.rs:37]: Refactored all these functions to return Result

File: src/auth/jwt/mod.rs [hunk 5]
```diff
@@ -0,0 +1,186 @@
...
+    #[error("Invalid JSON")]
+    Serialization,
+
+    #[error("Invalid issuer")]
+    IssuerInvalid,
+}
+
+/// Takes the result of a rsplit and ensure we only get 3 parts
+/// Errors if we don't
+macro_rules! expect_three {
```
[the remark this answers is not in the document]
Comment (Elyniss) [reply src/auth/jwt/mod.rs:43]: Refactored

File: terraform/docdb/main.tf (hunk 1 above)
Comment (Elyniss) [reply terraform/docdb/main.tf:7]: Updated

File: terraform/docdb/main.tf (hunk 2 above)
Comment (Elyniss) [reply terraform/docdb/main.tf:21]: I'm not sure. What do you suggest we do?

File: terraform/docdb/main.tf (hunk 3 above)
Comment (Elyniss) [reply terraform/docdb/main.tf:25]: I'm not sure. What do you suggest we do?

File: Cargo.toml [hunk 6]
```diff
@@ -1,19 +1,61 @@
 [package]
 name = "keyserver"
-version = "0.3.0"
+version = "1.0.0"
 edition = "2021"
-
-# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
+authors = [
+    "Derek Rein <derek@walletconnect.com>",
```
Comment (arein) [inline Cargo.toml:6]: awww
I did the least though.
Should be last!

File: src/auth/cacao/signature.rs [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+pub struct Signature {
+    pub t: String,
+    pub s: String,
+}
+
+impl Signature {
+    pub fn verify(&self, cacao: &Cacao) -> Result<bool, CacaoError> {
+        match self.t.as_str() {
+            "eip191" => Eip191.verify(&cacao.s.s, &cacao.p.address()?, &cacao.siwe_message()?),
+            // "eip1271" => Eip1271.verify(), TODO: How to accces our RPC?
```
Comment (arein) [inline src/auth/cacao/signature.rs:16]: via HTTP?

File: terraform/docdb/main.tf (hunk 1 above)
Comment (Elyniss) [reply terraform/docdb/main.tf]: After change, the `ci_terraform` workflow fails

Comment (xav) [reply terraform/docdb/main.tf]: The issue was something else. We don't always remove the Grafana API keys (like, when the CI fails) and expired keys still count in AWS quotas.
I cleaned up in Grafana and it should work now.
For ref: https://docs.aws.amazon.com/grafana/latest/userguide/AMG_quotas.html

File: terraform/docdb/main.tf (hunk 2 above)
Comment (xav) [reply terraform/docdb/main.tf:21]: Remove those two data and work directly with the resources they refer to.

**Review by xav — APPROVED** [review]
I agree with Derek, that's a HUUUUGE PR
I can't say I'm sure what I'm approving 😨 🥲

**Pull request merged**